db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/Predicate.java

```diff
@@ -92,6 +92,8 @@
 
     private ReferencedColumnsMap referencedColumns;
 
+    private HashMap<Optimizable, Double> scanSelectivityCache;
```

Comment:
Maybe it is better either to:
1. implement hash / equality explicitly for optimizable.
2. or, use a primitive type as a key where its hash / equality semantics are clear.

Reply:
That's a good point. I change the hash map to be table number -> scan selectivity.